source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -420,20 +420,42 @@ void DecodingProcessorState::clearWatermark() {
 }
 
 void DecodingProcessorState::clearRouteCache(const CommonResponse& common_response) {
-  if (!common_response.clear_route_cache()) {
-    return;
-  }
   if (filter_.config().isUpstream()) {
     filter_.stats().clear_route_cache_upstream_ignored_.inc();
     ENVOY_LOG(debug, "NOT clearing route cache. The filter is in upstream filter chain.");
     return;
   }
-  // Only clear the route cache if there is a mutation to the header and clearing is allowed.
-  if (filter_.config().disableClearRouteCache()) {
+
+  // If RouteCacheAction is configured with CLEAR.
+  if (filter_.config().routeCacheAction() ==
+      envoy::extensions::filters::http::ext_proc::v3::ExternalProcessor::CLEAR) {
+    if (common_response.has_header_mutation()) {
+      ENVOY_LOG(debug,
+                "Clearing route cache due to the filter is config with: RouteCacheconfig CLEAR");
+      decoder_callbacks_->downstreamCallbacks()->clearRouteCache();
+      return;
+    } else {
+      filter_.stats().clear_route_cache_ignored_.inc();
+      ENVOY_LOG(debug, "Filter RouteCacheAction is CLEAR. NO clearing route cache as no header "
+                       "mutations detected");
+      return;
+    }
+  }
+
+  if (!common_response.clear_route_cache()) {
+    return;
+  }
+
+  if (filter_.config().routeCacheAction() ==
```

Comment:
I think it might be simpler to normalize this configuration on ingestion. I.e. at config time, convert disable_clear_route_cache_ internally into a route cache action. That way there will be fewer branching conditions and need to worry about both older/newer way of configuring this internally.

Reply:
Merging `disable_clear_route_cache_` at FilterConfig() is a good idea.

It will save one `if statement` in my [suggested code example above](https://github.com/envoyproxy/envoy/pull/33830#discussion_r1583265577). i.e. we only need switch statement here.

Reply:
Good idea! done